### ISSUE TITLE
VoiceOver route announcement interferes with focused text field announcement

### ISSUE THREAD

**Issue by kf6gpe** [flutter/flutter#55306]
Internal: b/154684205

https://b.corp.google.com/issues/154566051.

Whenever the following steps are taken:

Focus on a text field
Activate a popup menu
Dismiss the popup menu
Upon dismissal, VoiceOver will announce the originally focused text filed, but the announcement will be interrupted by an announcement of the named route.

Minimal reproduction case:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Semantics(
        label: 'Voiceover Bug',
        explicitChildNodes: true,
        namesRoute: true,
        scopesRoute: true,
        child: Scaffold(
          appBar: AppBar(title: Text('First Route')),
          body: Column(
            children: [
              PopupMenuButton(
                itemBuilder: (_) => [PopupMenuItem(child: Text('Hi'))],
              ),
              TextField(),
            ],
          ),
        ),
      ),
    ),
  );
}
```

Demo video is attached. In the demo, notice that "Text Field VoiceOver Bug" is announced (VoiceOver bug is the name of the route). The correct announcement should be "Text Field, is editing, insertion point at start".

See internal bug for video.

**Comment by xster** [comment]
@goderbauer based on the description, not sure this is an engine bug (yet :)) 

Do you want to give it a first pass?

**Comment by goderbauer** [comment]
"VoiceOver Bug" is announced because the topmost route changes from the nameless pop-up route back to the Route that is names "VoiceOver Bug" by the Semantics widget. This is produced by this line:

https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm#L140

At the same time, iOS is trying to announce the new widget that is focused. iOS does not give us control over the order in which these are announced.

The only way I can see to fix the problem is to prevent one of the announcements and the only announcement we have control over is the route name one.

Possible solutions:

1) (in user code) On iOS it actually seems to be fairly uncommon to actually name a route. If you remove the namesRoute flag (maybe conditionally only when running on iOS), this problem goes away. 
2) (in the engine) In the iOS accessibility bridge we could change the logic that figures out whether a route name should be announced or not. I don't have a good idea for what that heuristic should be. Maybe they should just never be announced since it's rather uncommon on iOS?

**Comment by goderbauer** [comment]
Thinking about this a little more, I think 1) is the right way. If you don't want to announce your route's name when it becomes the topmost route then don't give it a name.

**Comment by tvolkert** [comment]
What customer response are we waiting for here?

**Comment by goderbauer** [comment]
We are waiting for the customer's responds to b/154566051#comment2.

**Comment by mehmetf** [comment]
From customer:

> Removing the route name works.  We will do that for this page as the page has a header that indicates the page name anyway.
